src/Helper/Dsn.php

```diff
@@ -6,9 +6,9 @@
 
 final class Dsn
 {
-    private ?string $dbname;
+    private string $dbname;
     private string $driver;
-    private ?string $host;
+    private string $host;
```

Comment:
It is OK not to speficy host. For example, SQLite uses `sqlite:/opt/databases/mydb.sq3` as connection string.

Reply:
I thought about it, but I still need to think about it.

It seems to me that we need to make the class abstract so that it does not know about any parameters other than the driver name and allow the drivers to extend this class for their databases.

How do you look at it?